### PR TITLE
nrop pkg: NUMAResourcesScheduler handler added

### PR DESCRIPTION
**Pull request by elenagerman**
nrop pkg: NUMAResourcesScheduler handler added

### CODE REVIEW

File: pkg/nrop/nrop_scheduler_test.go [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var (
+	defaultNROSchedulerName      = "numaresourcesscheduler"
+	defaultNROSchedulerNamespace = "openshift-numaresources"
+	defaultImageSpec             = "test-registry.redhat.io/openshift4/noderesourcetopology-scheduler-rhel9:v4.14.0"
```
Comment (cdvultur) [inline pkg/nrop/nrop_scheduler_test.go:18]: Can we change this to something generic ? In unit tests it should work without issue

**Review by cdvultur — CHANGES_REQUESTED** [review]
One comment on too specific variable name

File: pkg/nrop/nrop_scheduler_test.go (hunk 1 above)
Comment (elenagerman) [reply pkg/nrop/nrop_scheduler_test.go:18]: it is generic, but I can change it to the dummy format

**Review by cdvultur — APPROVED** [review]
/lgtm

**Comment by kononovn** [comment]
@elenagerman Please re-base

File: pkg/nrop/nrop_scheduler.go [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	nrosObj := &nropv1.NUMAResourcesScheduler{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name:      builder.Definition.Name,
+		Namespace: builder.Definition.Namespace,
+	}, nrosObj)
+
+	if err != nil {
+		return nil, err
```
Comment (kononovn) [inline pkg/nrop/nrop_scheduler.go:130]: Please add log message:
			"NUMAResourcesScheduler object %s does not exist in namespace %s",
			builder.Definition.Name, builder.Definition.Namespace)

**Review by kononovn — CHANGES_REQUESTED** [review]
1 minor comment to fix.

File: pkg/nrop/nrop_scheduler.go [hunk 3]
```diff
@@ -0,0 +1,280 @@
+package nrop
```
Comment (kononovn) [inline pkg/nrop/nrop_scheduler.go:1]: Please change names of files to: 
nropscheduler.go
nropscheduler_test.go

**Review by kononovn — CHANGES_REQUESTED** [review]
missed 1 detail. Please re-name files.

**Review by kononovn — APPROVED** [review]
/lgtm

**Pull request merged**